Libraries/LibWeb/WebGL/WebGLRenderingContextBase.h

```diff
@@ -117,6 +128,15 @@ class WebGLRenderingContextBase {
     Optional<ConvertedTexture> read_and_pixel_convert_texture_image_source(TexImageSource const& source, WebIDL::UnsignedLong format, WebIDL::UnsignedLong type, Optional<int> destination_width = OptionalNone {}, Optional<int> destination_height = OptionalNone {});
 
 protected:
+    static Vector<GLchar> null_terminated_string(StringView string)
+    {
+        Vector<GLchar> result;
+        for (auto c : string.bytes())
+            result.append(c);
+        result.append('\0');
+        return result;
+    }
```

Comment:
consider growing the vector to the target size and then copying the contents instead. (i believe you could re-use some existing copying mechanisms from the internals of `StringView`, `ByteString`, and `Vector`)